### PR TITLE
Add product unit filter to add night unit support

### LINKED ISSUE #320: Duration unit for Accommodation Booking products should be "Nights" not "Blocks"
#### Affected ticket(s)

5808811-zen

#### What I expected

Given that these products are booked for a number of nights, I expected that the message about minimum duration would reference the `Duration_unit` for an accommodations product as "Nights"

#### What happened instead

Instead we get the following message "The minimum duration is 3 **blocks**"

#### Steps to reproduce the issue

- Create an Accommodations Booking Product.
- Set the minimum duration to two or more nights, the maximum to a short range as well.
- Navigate to the front end and view your product
- Select a number of days under or over the allowed range
- Note that the message refers to the `duration_unit` as "Blocks" not nights.

The user in 5808811-zen put it this way

> If you select a start day followed be the next day appearing as two “blocks” then the calendar still insists you need to select two blocks. It should perhaps say 2 “nights. This appears as a 3 block selection.

I searched and see no user-facing way to set the `duration_unit` to `nights`, so assume this is hard-coded and must be addressed by a PR.

-------------------

- [ ] Issue assigned to next milestone.
- [ ] Issue assigned a priority (will be assessed by maintainers).

### PR DESCRIPTION
**Pull request by kirtangajjar**
### Changes proposed in this Pull Request:

Filter product unit to add night unit support.

Closes #320.

### How to test the changes in this Pull Request:

1. Create an Accommodations Booking Product.
2. Set the minimum duration to two or more nights, the maximum to a short range as well.
3. Navigate to the front end and view your product
4. Select a number of days under or over the allowed range
5. Note that the message refers to the duration_unit as "Nights", not "Blocks".

### Changelog entry

> Dev - Add product unit filter, `wc_bookings_product_duration_fallback`, to add night unit support.

### CODE REVIEW

**Comment by dkotter** [comment]
@kirtangajjar Is this one ready for code review?

File: includes/admin/class-wc-accommodation-booking-admin-panels.php [hunk 1]
```diff
@@ -24,6 +25,22 @@ public function __construct() {
...
+	 * @param string $duration_unit_default Default fallback duration
+	 * @param string $duration_unit         Current duration unit
+	 * @param int    $duration              Duration of booking
+	 *
+	 * @return string
+	 */
+	public function get_product_duration( $duration_unit_default, $duration_unit, $duration ) {
+		if ( 'night' === $duration_unit ) {
+			return _n( 'night', 'nights', $duration, 'woocommerce-accommodation-bookings' );
+		}
```
Comment (faisal-alvi) [inline includes/admin/class-wc-accommodation-booking-admin-panels.php:40]: At this point, we would get the value already filtered through the `_n` function from [here](https://github.com/woocommerce/woocommerce-bookings/blob/d37b2fc423aa077ed3ef154b1ebe8f1587a1f9c6/includes/data-objects/class-wc-product-booking.php#L2421).

However, we're still filtering back through the `_n` with the new values (i.e. `night`, `nights`). But seems reasonable.

**Review by faisal-alvi — CHANGES_REQUESTED** [review]
Added feedback at https://github.com/woocommerce/woocommerce-bookings/pull/3564#pullrequestreview-1378867304, which would require changes in this PR too.

File: includes/admin/class-wc-accommodation-booking-admin-panels.php (hunk 1 above)
Comment (kirtangajjar) [reply includes/admin/class-wc-accommodation-booking-admin-panels.php:40]: Yes, that's true; we could've saved a function call if we tried to filter just those two values but for the sake of simplicity of implementation, I liked using this approach.

**Review by faisal-alvi — APPROVED** [review]

**Comment by ankitguptaindia** [comment]
**QA/Test Report-** ✅ 
------------

**Testing Environment -** 

<Details>

* WordPress: 6.2 
* Theme: Storefront Version: 4.2.0
* WooCommerce - Version 7.7.0
* PHP: 8.0.22
* Web Server: Nginx
* Browser: Chrome - Version 113
* OS: macOS Monterey V 12.3.1
* Git Branch:  fix/320 & fix/accomodation-booking-320  of https://github.com/woocommerce/woocommerce-bookings/pull/3564 

</Details>


**Test Results -** 

Upon selecting a number of days under or over the allowed range, the system displayed a message referring to the duration unit as "Nights," not "Blocks." The message accurately reflects the configured duration unit.

Additionally, the test was performed on other places where the duration appears, such as the checkout page, order page, emails, and booking page. Newly added/updated string found as translatable as well. 


**Functional Demo /  Screencast -**

Screenshot 2023-05-16 at 6 10 13 PM

Screenshot 2023-05-16 at 6 10 36 PM






**Next Step-** Ready to Code Review(Woo)

**Review by ankitguptaindia — APPROVED** [review]

**Comment by ankitguptaindia** [comment]
@dkotter Does this PR require assigning Woo team to review? this looks like a simple change.

**Review by faisal-alvi — APPROVED** [review]

**Comment by faisal-alvi** [comment]
@ankitguptaindia yes we can move this straight to UAT once QA test passes.

**Comment by ankitguptaindia** [comment]
**Regression+Smoke Test Report-** ✅ 
------------

**Testing Environment -** 


<Details>

* WordPress: 6.2.2
* Theme: Storefront Version: 4.2.0
* WooCommerce - Version 7.7.0
* PHP: 8.0.22 
* Web Server: Nginx
* Browser: Chrome - Version 113


</Details>


Tested with Archive File created via 

```php woorelease.phar build https://github.com/woocommerce/woocommerce-accommodation-bookings/tree/smoke-testing```

- Composer version - 2.3.5
- npm version - 6.14.14
- node version - 14.17.5

#### Please note that this plugin is tested with the build created by specified versions of Composer, Node, and NPM. 



**Status-** Ready to Merge 🚀

**Pull request merged**